fix crash enabling PFL after deleting EffectState while PFL is disabled

Be-ing opened the pull request:
fixing https://bugs.launchpad.net/mixxx/+bug/1741213

Be-ing committed:
move kEffectDebugOutput constant to a more appropriate place

Be-ing committed:
add more effect debugging output

Be-ing committed:
fix crash enabling PFL after an EffectState is deleted

Be-ing commented:
Ping. Can we prioritize this critical bug fix?

uklotzde commented on src/effects/defs.h:
```diff
@@ -28,6 +28,11 @@ enum class EffectChainInsertionType {
+// NOTE: Setting this to true will enable string manipulation and calls to
+// qDebug() in the audio engine thread. That may cause audio dropouts, so only
+// enable this when debugging the effects system.
+const bool kEffectDebugOutput = false;
```
A constexpr might be more suitable to enforce evaluation (and optimization) at compile time

uklotzde commented:
The crash is fixed, but replaced by another minor bug: When re-routing the deck to the effect unit with the effect still activated, it is not applied. You need to deactivate and activate the effect once, before it gets applied again.

Be-ing committed:
fix effects not being processed after toggling routing switch twice

EngineEffectChain should be responsible for managing routing
switch states, not EngineEffect.

Be-ing committed:
use constexpr to allow for compile time optimization

uklotzde commented:
LGTM. Just waiting for the builds to finish.
